tools/generatemodel.py

```diff
@@ -5,6 +5,54 @@
 
 import yaml
 
+import os
+import requests
+
+
+# Function to fetch the latest schema version from GitHub
+def fetch_latest_schema_version():
```

Comment:
it may be better to parameterize this function to allow a specific version to be generated and only generate the latest version if the parameter is not set:

`def fetch_schema_version(version_tag=None): ...`

Reply:
and also make sure, the user provided a version_tag => 1.8.0 as the schema.json was only introduced beginning with 1.8.0